src/Symfony/Component/HttpKernel/EventListener/AbstractSessionListener.php

```diff
@@ -200,10 +200,14 @@ public function onKernelResponse(ResponseEvent $event)
         }
 
         if ($autoCacheControl) {
+            $maxAge = 0;
+            if (!$response->headers->hasCacheControlDirective('public') && (bool) $response->getMaxAge()) {
```

Comment:
I'd suggest this CS instead:
```diff
         if ($autoCacheControl) {
+            $maxAge = $response->headers->hasCacheControlDirective('public') ? 0 : (int) $response->getMaxAge();
             $response
-                ->setExpires(new \DateTime())
+                ->setExpires(new \DateTimeImmutable('+'.$maxAge.' seconds'))
```

But LGTM otherwise thanks!

(please allow edits from maintainers, that makes it easier to deal with contributions on our side)

Reply:
Hi @nicolas-grekas ,
   I have applied the suggested CS, looks better now.
I can't allow edits for maintainers... I think it's because I opened the PR from an organization repository and that option is not available in that case (please correct me if I'm wrong).